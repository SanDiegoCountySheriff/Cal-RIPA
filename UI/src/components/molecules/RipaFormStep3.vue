<template>
  <v-form ref="stepForm" lazy-validation>
    <ripa-stop-reason
      v-model="model"
      :isOnlineAndAuthenticated="isOnlineAndAuthenticated"
      :last-reason="lastReason"
      :loading-pii="loadingPii"
      :statutes="getStatutes"
      :on-open-favorites="onOpenFavorites"
      :on-open-statute="onOpenStatute"
      :on-save-favorite="onSaveFavorite"
      @pii-check="handlePiiCheck"
    ></ripa-stop-reason>

    <v-spacer></v-spacer>

    <template v-if="!isFormValid">
      <ripa-alert alert-type="error">
        Oops, you may have missed something! Please review your selections
        above.
      </ripa-alert>
    </template>

    <div class="tw-flex tw-mt-8 tw-justify-center">
      <v-btn
        outlined
        color="primary"
        class="tw-mr-2"
        :disabled="isBackNextDisabled"
        @click="handleBack"
      >
        Back
      </v-btn>
      <v-btn outlined color="error" class="tw-mr-2" @click="handleCancel">
        Cancel
      </v-btn>
      <v-btn
        color="primary"
        :disabled="isBackNextDisabled"
        @click="handleStep3Next"
      >
        Next
      </v-btn>
    </div>

    <ripa-confirm-dialog
      :show-dialog="showConfirmDialog"
      title="Confirm Continue"
      subtitle="This page may contain personally identifying information. Are you sure you want to continue?"
      :on-close="handleCloseDialog"
      :on-confirm="handleConfirm"
    >
    </ripa-confirm-dialog>
  </v-form>
</template>

<script>
import RipaAlert from '@/components/atoms/RipaAlert'
import RipaConfirmDialog from '@/components/atoms/RipaConfirmDialog'
import RipaStopReason from '@/components/molecules/RipaStopReason'
import RipaFormStepMixin from '@/components/mixins/RipaFormStepMixin'

export default {
  name: 'ripa-form-step3',

  mixins: [RipaFormStepMixin],

  components: { RipaAlert, RipaConfirmDialog, RipaStopReason },

  methods: {
    handleStep3Next() {
      const piiFound = this.viewModel.stopReason?.reasonForStopPiiFound || false
      if (piiFound) {
        this.showConfirmDialog = true
      } else {
        this.handleNext()
      }
    },

    handleCloseDialog() {
      this.showConfirmDialog = false
    },

    handlePiiCheck({ source, value }) {
      this.$emit('pii-check', { source, value })
    },
  },

  computed: {
    getStatutes() {
      return this.statutes
    },
  },

  props: {
    isOnlineAndAuthenticated: {
      type: Boolean,
      default: false,
    },
    lastReason: {
      type: Object,
      default: () => {},
    },
    onOpenFavorites: {
      type: Function,
      required: true,
    },
    onSaveFavorite: {
      type: Function,
      required: true,
    },
  },
}
</script>
