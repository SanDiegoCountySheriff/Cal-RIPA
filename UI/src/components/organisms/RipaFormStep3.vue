<template>
  <div>
    <v-form ref="step3" lazy-validation>
      <ripa-stop-reason
        ref="stopReason1"
        v-model="stopReason1"
      ></ripa-stop-reason>
      <!-- <ripa-stop-reason
        ref="stopReason2"
        v-model="stopReason2"
      ></ripa-stop-reason> -->
    </v-form>
    <v-btn color="primary" class="tw-mr-4" @click="submit"> Submit </v-btn>
    <v-btn color="error" class="tw-mr-4" @click="reset"> Reset </v-btn>
  </div>
</template>

<script>
import RipaStopReason from '@/components/molecules/RipaStopReason'

export default {
  name: 'ripa-form-step3',

  components: { RipaStopReason },

  data() {
    return {
      stopReason1: {
        name: this.value.name1,
      },
      stopReason2: {
        name: this.value.name2,
      },
    }
  },

  methods: {
    submit() {
      const isValid = this.$refs.step3.validate()
      if (!isValid) {
        return
      }
      const stopReason1Form = this.$refs.stopReason1
      const stopReason2Form = this.$refs.stopReason2
      this.$emit('input', {
        stopReason1: {
          name: stopReason1Form.name,
          year: stopReason1Form.email,
          select: stopReason1Form.select,
          checkbox: stopReason1Form.checkbox,
        },
        stopReason2: {
          name: stopReason2Form.name,
          year: stopReason2Form.email,
          select: stopReason2Form.select,
          checkbox: stopReason2Form.checkbox,
        },
      })
    },

    reset() {
      this.$refs.step3.reset()
    },
  },

  props: {
    value: {
      type: Object,
      default: () => {},
    },
  },
}
</script>
