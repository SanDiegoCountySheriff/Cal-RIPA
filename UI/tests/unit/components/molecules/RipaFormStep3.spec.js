import RipaFormStep3 from '@/components/molecules/RipaFormStep3.vue'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Form Step 3', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaFormStep3, {
      vuetify,
      propsData: {
        ...propsData,
        onOpenFavorites: jest.fn(),
        onSaveFavorite: jest.fn(),
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaFormStep3, {
      vuetify,
      propsData: {
        onOpenFavorites: jest.fn(),
        onSaveFavorite: jest.fn(),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
