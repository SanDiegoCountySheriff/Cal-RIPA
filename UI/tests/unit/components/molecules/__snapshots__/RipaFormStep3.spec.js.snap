// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Form Step 3 should match snapshot 1`] = `
<form novalidate="novalidate" class="v-form" onopenfavorites="function () {
        return fn.apply(this, arguments);
      }" onsavefavorite="function () {
        return fn.apply(this, arguments);
      }">
  <div class="ripa-stop-reason tw-pb-4">
    <div class="ripa-form-header">
      <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
        <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Reason for Stop</span></div>
      </div>
    </div>
    <div class="container">
      <div class="row no-gutters">
        <div class="tw-pr-2 col-sm-6 col-md-6 col-12">
          <div class="tw-mr-2 tw-mt-4"><button type="button" class="tw-w-full v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary"><span class="v-btn__content">
            Open Favorites
          </span></button></div>
        </div>
        <div class="col-sm-6 col-md-6 col-12">
          <div class="tw-mr-2 tw-mt-4"><button type="button" class="tw-w-full v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary"><span class="v-btn__content">
              Save Reason
            </span></button></div>
        </div>
      </div>
      <div class="row">
        <!---->
      </div>
      <div class="row">
        <div class="text-center col">
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
      </div>
      <div class="row no-gutters">
        <div class="col-sm-12 col-12">
          <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--solo-flat v-select">
            <div class="v-input__control">
              <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-12" class="v-input__slot">
                <div class="v-select__slot"><label for="input-12" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Reason</label>
                  <div class="v-select__selections">
                    <div class="v-select__selection v-select__selection--comma">Traffic Violation</div><input required="required" id="input-12" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--clear"><button type="button" aria-label="clear icon" tabindex="-1" class="v-icon notranslate v-icon--link mdi mdi-close theme--light"></button></div>
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                  </div><input type="hidden" value="1">
                </div>
                <div class="v-menu">
                  <!---->
                </div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <!---->
          <div class="ripa-radio-group">
            <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
              <div class="v-input__control">
                <div class="v-input__slot" style="height: auto;">
                  <div role="radiogroup" aria-labelledby="input-19" class="v-input--radio-group__input">
                    <div class="v-radio theme--light v-item--active">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"></i><input aria-checked="true" id="input-20" role="radio" type="radio" name="radio-19" value="1">
                        <div class="v-input--selection-controls__ripple primary--text"></div>
                      </div><label for="input-20" class="v-label theme--light" style="left: 0px; position: relative;">Moving Violation</label>
                    </div>
                    <div class="v-radio theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-22" role="radio" type="radio" name="radio-19" value="2">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-22" class="v-label theme--light" style="left: 0px; position: relative;">Equipment Violation</label>
                    </div>
                    <div class="v-radio theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-24" role="radio" type="radio" name="radio-19" value="3">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-24" class="v-label theme--light" style="left: 0px; position: relative;">Non-moving Violation, including Registration Violation</label>
                    </div>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
            <!---->
          </div>
          <div class="tw-mt-2"></div>
          <div class="v-input tw-my-4 theme--light v-text-field v-text-field--solo-flat v-select v-autocomplete">
            <div class="v-input__control">
              <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-29" class="v-input__slot">
                <div class="v-select__slot"><label for="input-29" class="v-label theme--light" style="left: 0px; position: absolute;">Offense Code</label><input id="input-29" type="text" autocomplete="off">
                  <div class="v-input__append-inner">
                    <div></div>
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                  </div><input type="hidden" value="1">
                </div>
                <div class="v-menu">
                  <!---->
                </div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper">
                    <div class="v-messages__message">Select 1 Offense Code (required)</div>
                  </transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <!---->
          <!---->
          <!---->
          <div class="ripa-label tw-font-bold tw-my-4"><span>
    -- and --
  </span></div>
          <!---->
          <div class="ripa-text-input" persistent-hint="">
            <!---->
            <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-text-field__slot"><label for="input-37" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Brief Explanation</label><input id="input-37" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
  <div class="spacer"></div>
  <!---->
  <div class="tw-flex tw-mt-8 tw-justify-center"><button type="button" class="tw-mr-2 v-btn v-btn--outlined theme--light v-size--default primary--text"><span class="v-btn__content">
      Back
    </span></button> <button type="button" class="tw-mr-2 v-btn v-btn--outlined theme--light v-size--default error--text"><span class="v-btn__content">
      Cancel
    </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">
      Next
    </span></button></div>
  <div class="v-dialog__container">
    <!---->
  </div>
</form>
`;
